lib/bolt/bolt_option_parser.rb

```diff
@@ -389,15 +383,9 @@ def initialize(options)
       define('--run-as USER', 'User to run as using privilege escalation') do |user|
         @options[:'run-as'] = user
       end
-      define('--sudo-password [PASSWORD]',
```

Comment:
I think you can just keep this "optional" with 
```
      define('-p', '--password [PASSWORD]',
             'Password to authenticate with') do |password|
          if password.nil?
            raise Bolt::CLIError,
                  "DEPRECATION MESSAGE: NO MORE PASSWORD PROMt"
          else
            @options[:password] = password
          end
      end
```
and then raise cli error with deprecation message.

Reply:
Updated.